File: loan-repayment-java/src/test/java/com/loan/repayment/factory/RepaymentPlanResponseFactory.java
package com.loan.repayment.factory;

import com.loan.repayment.entity.RepaymentPlanResponse;
import com.loan.repayment.factory.util.JsonUtil;

/**
 * 
 * @author Ramachandra
 *
 */
public class RepaymentPlanResponseFactory {

    /**
     * This method creates the RepaymentPlanResponse for the 6 months repayment
     * scheduled plan
     * 
     * @return RepaymentPlanResponse
     */
    public static RepaymentPlanResponse createRepaymentPlanResponse() {
        return (RepaymentPlanResponse) new JsonUtil().readFileAndGetData("repaymentscheduleresponse6months.json",
                RepaymentPlanResponse.class);
    }

    /**
     * This method creates the RepaymentPlanResponse for the one month repayment
     * scheduled plan
     * 
     * @return RepaymentPlanResponse
     */
    public static RepaymentPlanResponse createRepaymentPlanResponseSingleMonth() {
        return (RepaymentPlanResponse) new JsonUtil().readFileAndGetData("repaymentscheduleresponse1month.json",
                RepaymentPlanResponse.class);
    }

}

File: loan-repayment-kotlin/src/test/kotlin/com/loan/repayment/factory/RepaymentPlanResponseFactory.kt
package com.loan.repayment.factory

import com.loan.repayment.entity.RepaymentPlanResponse
import com.loan.repayment.factory.util.JsonUtil

/**
 *
 * @author Rama chandra
 */
object RepaymentPlanResponseFactory {
    /**
     * This method creates the RepaymentPlanResponse for the 6 months repayment
     * scheduled plan
     *
     * @return RepaymentPlanResponse
     */
    fun createRepaymentPlanResponse() : RepaymentPlanResponse {
        return JsonUtil().readFileAndGetData("repaymentscheduleresponse6months.json" ,
                RepaymentPlanResponse::class.java) as RepaymentPlanResponse
    }

    /**
     * This method creates the RepaymentPlanResponse for the one month repayment
     * scheduled plan
     *
     * @return RepaymentPlanResponse
     */
    fun createRepaymentPlanResponseSingleMonth() : RepaymentPlanResponse {
        return JsonUtil().readFileAndGetData("repaymentscheduleresponse1month.json" ,
                RepaymentPlanResponse::class.java) as RepaymentPlanResponse
    }
}